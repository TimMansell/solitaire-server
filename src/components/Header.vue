<template>
  <header class="header">
    <Controls />
    <Stats />
  </header>
</template>

<script>
import Controls from '@/components/Controls.vue';
import Stats from '@/components/Stats.vue';

export default {
  name: 'Header',
  components: {
    Controls,
    Stats,
  },
};
</script>

<style scoped lang="scss">
.header {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  position: sticky;
  top: 0;
  z-index: var(--z-header);
  text-shadow: 0 1px var(--col-primary-dark-2);
  padding: var(--pd-sm);
  background: var(--bg-primary);
  font-size: var(--font-size-sm);

  @media (min-width: $bp-sm) {
    flex-direction: row;
    border-bottom: 1px solid var(--bdr-secondary);
    box-shadow: 0 1px rgba($col-secondary, 0.05);
    margin-bottom: var(--mg-sm);
  }

  @media (min-width: $bp-md) {
    font-size: var(--font-size);
    margin-bottom: var(--mg-md);
  }
}
</style>
