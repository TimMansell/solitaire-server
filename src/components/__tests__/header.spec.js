import { shallowMount } from '@vue/test-utils';
import Header from '@/components/Header.vue';

describe('Header.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Header);

    expect(wrapper).toMatchSnapshot();
  });
});
