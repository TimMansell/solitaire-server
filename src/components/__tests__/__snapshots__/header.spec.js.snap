// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue matches snapshot 1`] = `
<header class="header">
  <controls-stub></controls-stub>
  <gamestats-stub></gamestats-stub>
</header>
`;
